hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -2088,25 +2165,20 @@ public boolean isAtomicRenameKey(String key) {
   RenameAtomicityUtils.RedoRenameInvocation getRedoRenameInvocation(final TracingContext tracingContext) {
     return new RenameAtomicityUtils.RedoRenameInvocation() {
       @Override
-      public void redo(final Path destination, final List<Path> sourcePaths,
-          final List<String> destinationSuffix)
+      public void redo(final Path destination, final Path src)
           throws AzureBlobFileSystemException {
-        for (int i = 0; i < sourcePaths.size(); i++) {
-          try {
-            final Path destinationPath;
-            if (destinationSuffix.get(i).isEmpty()) {
-              destinationPath = destination;
-            } else {
-              destinationPath = new Path(destination, destinationSuffix.get(i));
-            }
-            renameBlob(destinationPath, tracingContext, sourcePaths.get(i));
-          } catch (AbfsRestOperationException ex) {
-            if (ex.getStatusCode() == HttpURLConnection.HTTP_NOT_FOUND) {
-              continue;
-            }
-            throw ex;
-          }
-        }
+
+        ListBlobQueue listBlobQueue = new ListBlobQueue(null);
```

Comment:
Can we avoid passing null as argument to the constructor ?

Reply:
Taken.